README.md

```diff
@@ -1,3 +1,41 @@
-# ecomm
-
+# Ecomm
 Projeto de Ecommerce criado durante o programa LevelUp da Alura
+
+## Table of contents
+
+- [Twelve-Factor App](#twelve-factor-app)
+- [Microservices Patterns](#microservices-patterns)
+
+## Twelve-Factor App
+
+| Factor | Description | Status |
+|--------|-------------|--------|
+| 1. Codebase | One codebase tracked in revision control, many deploys | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 2. Dependencies | Explicitly declare and isolate dependencies | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 3. Config | Store config in the environment | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 4. Backing services | Treat backing services as attached resources | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 5. Build, release, run | Strictly separate build and run stages | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 6. Processes | Execute the app as one or more stateless processes | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 7. Port binding | Export services via port binding | <img alt="Pass" src="https://img.shields.io/badge/-Pass-brightgreen"> |
+| 8. Concurrency | Scale out via the process model | <img alt="Fail" src="https://img.shields.io/badge/-Fail-red"> |

```

Comment:
Pq vc considerou este quesito não foi atendido? 
***The twelve-factor app is completely self-contained and does not rely on runtime injection of a webserver into the execution environment to create a web-facing service. The web app exports HTTP as a service by binding to a port, and listening to requests coming in on that port.***